src/v/cluster/inventory_service.cc

```diff
@@ -152,15 +152,20 @@ inventory_service::inventory_service(
   , _should_create_report_config{should_create_report_config} {}
 
 ss::future<> inventory_service::start() {
-    if (ss::this_shard_id() != 0) {
+    if (ss::this_shard_id() != shard_id) {
         co_return;
     }
 
-    auto rtc = make_rtc(_as);
-
-    vlog(cst_log.info, "Attempting to create inventory configuration");
+    vlog(
+      cst_log.info,
+      "starting inventory download service, should create config: {}, check "
+      "interval: {}",
+      _should_create_report_config,
+      _inventory_report_check_interval);
 
     if (_should_create_report_config) {
+        vlog(cst_log.info, "Attempting to create inventory configuration");
+        auto rtc = make_rtc(_as);
```

Comment:
nit: should we make similar changes to check_for_current_inventory()?

Seems like maybe this method could be pulled into some inventory_service::maybe_create_inventory_confg() that does everything in this if block (but also considers _try_creating_inv_config). The if block in check_for_current_inventory() could use this too

Reply:
added call to `make_rtc` - looking into the other refactoring

Reply:
refactored into a common method